src/main/java/org/isf/menu/gui/MainMenu.java

```diff
@@ -124,6 +125,7 @@ public MainMenu() {
 		myFrame = this;
 
 		GeneralData.getGeneralData();
+		MessageBundle.getBundle();
```

Comment:
What is the purpose of this statement; it returns a `ResourceBundle` but it already has to exist and this statement is not even saving the returned value?

I will also point out that `GeneralData.getGeneralData()` returns a singleton `GeneralData` that is never assigned in the 11 occurrences in OH.    And then the fact that `getGeneralData` is responsible for initializing the ResourceBundle is also less than obvious.

If the methods are not going to do anything with the returned value then instead of `getGeneralData()` it should be something like `initGeneralData()`.   And instead of burying the ResourceBundle init maybe the init routine should be called directly.

Reply:
true, for MessageBundle it can be initialized with the `initialize()` method; anyway, both cases "getter" is an heritage of previous architecture. According to GOF, we could rename all getters of singleton classes into `getInstance()`, in this way whereas the returned object is used or not the call is more explicit, wdyt?

Reply:
Yes it makes it more meaningful but just barely IMHO as there is still no *getting*.   All the method is doing is initializing a global object that is accessed through `public static` variables like `GeneralData.OPDEXTENDED`; as it now written there is no reason ever to get a handle to the instance.

Reply:
Fair enough, but this let me think also another thing: having public static variables allow any piece of code to access them even if not initialized, without knowing if they host the intended values or not, and even changing them afterward; this can be fine to programmatically change some behviors but it can lead to security problem if we see at SINGLEUSER variable in GeneralData (which should be final once intialized...). All the changes in this PR were meant to accomodate actual architecture, but maybe we have to rethink it...

Reply:
To be more clear I've pushed an update: 1) both initialize() and getter() 2) private final SINGLEUSER with getter;
Could it be a valid intermediate solution for the above?

Reply:
Looks better thanks; and +1 as an intermediate solution.

Reply:
Thanks to you also!